front/lib/api/assistant/conversation.ts

```diff
@@ -1703,3 +1724,53 @@ async function isMessagesLimitReached({
 
   return remaining <= 0;
 }
+
+async function storeContentFragmentText({
+  workspaceId,
+  conversationId,
+  messageId,
+  content,
+}: {
+  workspaceId: string;
+  conversationId: string;
+  messageId: string;
+  content: string;
+}): Promise<string | null> {
+  if (content === "") {
+    return null;
+  }
+
+  const { filePath, fileUrl } = contentFragmentUrl({
+    workspaceId,
+    conversationId,
+    messageId,
+    contentFormat: "text",
+  });
+  const storage = new Storage({
+    keyFilename: appConfig.getServiceAccount(),
+  });
+
+  const bucket = storage.bucket(gcsConfig.getGcsPrivateUploadsBucket());
+  const gcsFile = bucket.file(filePath);
+
+  await gcsFile.save(content, {
+    contentType: "text/plain",
+  });
+
+  return fileUrl;
+}
+
+async function countTokens(content: string): Promise<number> {
```

Comment:
I'd challenge doing this here, our UI is already a bit sluggish, and this might slow it down even more. Do we have a particular reason for storing the token count on the content fragment? Also, do we have any idea how long this API call might take?

Reply:
Thx for commenting, worth discussing indeed.

Looping in @spolu as DRI / framer on the new ContentFragment model

The call to /tokenize is most often a few miliseconds, see [here](https://app.datadoghq.eu/logs?query=tokenize%20&cols=service%2C%40workflowName%2C%40activityName%2C%40error.__is_dust_error&fromUser=true&index=%2A&messageDisplay=inline&refresh_mode=sliding&sp=&storage=hot&stream_sort=%40duration%2Cdesc&viz=stream&from_ts=1710083152274&to_ts=1710342352274&live=true) but might be heavier on long text though (we accept up to ~2MB text, long term we could get fragments with hundreds of K of tokens)

It's true it's another potential failure/latency point, and we don't use this count at the time so maybe worth making it async or skipping it entirely -- will check how long the call is on large text, this may help decide

Additional topic: the "model" used for the call to tokenize, see comment in `countTokens`. IMHO fine for now but worth mentioning

Reply:
Tokenizing the largest file we allow for upload (actually 1Mb of text) => takes 2s on laptop, so latency we cannot ignore

We can move it to async or skip filling the field for now

Reply:
Since we're not using it for now let's just kill the whole thing if we fear it might introduce latency.

We will always be able to backfill and start counting as needed quite easily :+1:

Reply:
Done, will remove the field too